pkg/minikube/node/start.go

```diff
@@ -180,12 +181,33 @@ func Start(starter Starter, apiServer bool) (*kubeconfig.Settings, error) {
 		}
 	}
 
+	if err := restartKubeletIfNeeded(starter); err != nil {
+		return nil, errors.Wrap(err, "restarting kubelet")
+	}
+
 	wg.Wait()
 
 	// Write enabled addons to the config before completion
 	return kcs, config.Write(viper.GetString(config.ProfileName), starter.Cfg)
 }
 
+func restartKubeletIfNeeded(starter Starter) error {
+	if !starter.PreExists {
+		return nil
+	}
+	if !driver.IsKIC(starter.Host.DriverName) {
+		return nil
+	}
+	// on soft start and kic, we need to restart kubelet to ensure that it picks up the correct node ip
```

Comment:
I used to use "soft start" for starting a "running" minikube. I would use second-start or a different word that reflects starting a stopped minikube.

Reply:
this code is now removed